src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting_NetFx/CodeDirectiveFormattingTest.cs

```diff
@@ -37,6 +37,44 @@ public interface Bar
                     """);
     }
 
+    [Fact]
+    public async Task FormatsCodeBlockDirectiveWithCRLFLineEnding()
+    {
+        await RunFormattingTestAsync(
+            input:
+                "@code {\r\n" +
+                " public class Foo{}\r\n" +
+                "        public interface Bar {\r\n" +
+                "}\r\n" +
+                "}",
+            expected:
+                "@code {\r\n" +
+                "    public class Foo { }\r\n" +
+                "    public interface Bar\r\n" +
+                "    {\r\n" +
+                "    }\r\n" +
+                "}");
```

Comment:
Consider adding test helpers to add the line breaks for you. Something like these would make it a little less error prone to write new tests.

```C#
public static string CodeFromLines(string lineEnding, params string[] lines)
{
    return string.Join(lineEnding, lines);
}
```

Then, the tests can be written by passing in the line text without line endings like so:

```suggestion
        await RunFormattingTestAsync(
            input: CodeFromLines("\r\n",
                "@code {",
                " public class Foo{}",
                "        public interface Bar {",
                "}",
                "}"),
            expected: CodeFromLines("\r\n",
                "@code {",
                "    public class Foo { }",
                "    public interface Bar",
                "    {",
                "    }",
                "}");
```

Even better, since this is test code, efficiency is less worrisome than in production code. So, you could write a helper that uses a `SourceText` to parse the lines and use that to build up a new string with different line endings. Something like the following would do the trick (Note: untested code!).

```C#
public static string NormalizeLineEndings(string lineEnding, string code)
{
    using var _ = StringBuilderPool.GetPooledObject(out var builder);

    var text = SourceText.From(code);

    foreach (var line in text.Lines)
    {
        builder.Append(text.GetSubTextString(line.Span));

        if (line.End < text.Length)
        {
            builder.Append(lineEnding);
        }
    }

    return builder.ToString();
}
```

Then, the tests could continue using raw string literals like so:

```suggestion
        await RunFormattingTestAsync(
            input: NormalizeLineEndings("\r\n", """
                @code {
                 public class Foo{}
                        public interface Bar {
                }
                }
                """,
            expected: NormalizeLineEndings("\r\n", """
                @code {
                    public class Foo { }
                    public interface Bar
                    {
                    }
                }
                """);
```

Even, even better, the test helper could be extension method on string.

```C#
public static string ChangeLineEndingsTo(this string code, string lineEnding);
```